packages/trader/src/Modules/MT5/Components/mt5-poa.jsx

```diff
@@ -0,0 +1,453 @@
+                                values={{
+                                    field_name: mappedKey[key],
+                                }}
+                                options={{ interpolation: { escapeValue: false } }}
+                            />;
+                    }
+                }
+            });
+
+        return errors;
+    };
+
+    handleCancel = (values) => {
+        this.props.onSave(this.props.index, values);
+        this.props.onCancel();
+    };
+
+    onFileDrop = (document_file, file_error_message, setFieldTouched, setFieldValue) => {
+        setFieldTouched('document_file', true);
+        setFieldValue('document_file', document_file);
+        this.setState({ document_file, file_error_message });
+    };
+
+    proceed = () => {
+        this.props.onSubmit(this.props.index, ...this.state);
+    };
+
+    onSubmit = async (values, actions) => {
+        const { document_file, ...uploadables } = values;
+        actions.setSubmitting(true);
+        const data = await WS.setSettings(uploadables);
+        if (data.error) {
+            this.setState({
+                form_error: data.error.message,
+            });
+
+            actions.setSubmitting(false);
+            return;
+        }
+        const { error, get_settings } = await WS.authorized.storage.getSettings();
+        if (error) {
+            this.setState({
+                form_error: error.message,
+            });
+            return;
+        }
+
+        // Store newly stored values in the component.
+        const {
+            address_line_1,
+            address_line_2,
+            address_city,
+            address_state,
+            address_postcode,
+        } = get_settings;
+
+        this.setState({
+            address_line_1,
+            address_line_2,
+            address_city,
+            address_postcode,
+            address_state,
+            form_error: '',
+        });
+
+        try {
+            const api_response = await this.file_uploader_ref.current.upload();
+            if (api_response.warning) {
+                this.setState({
+                    form_error: api_response.warning,
+                });
+                actions.setSubmitting(false);
+                return;
+            }
+            const { error: e, get_account_status } = await WS.authorized.storage.getAccountStatus();
+            if (e) {
+                this.setState({
+                    form_error: error.message,
+                });
+                actions.setSubmitting(false);
+                return;
+            }
+            const { identity } = get_account_status.authentication;
+            const has_poi      = !(identity && identity.status === 'none');
+            if (has_poi) {
+                this.proceed();
+            } else {
+                this.setState({
+                    form_error: localize('Proof of identity failed. redirecting you back to previous step.'),
```

Comment:
Is this text accurate as per requirements? Mainly the capitalisation.

Reply:
I'll check with the copy-write team.

Reply:
Replaced it with a new one.